File: app/src/main/java/me/zipi/navitotesla/api/TMapApi.java
package me.zipi.navitotesla.api;

import me.zipi.navitotesla.BuildConfig;
import me.zipi.navitotesla.model.TMap;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Query;

public interface TMapApi {
    @GET("/tmap/pois?version=1&appKey=" + BuildConfig.TMAP_API_KEY)
    @Headers({
            "Accept: application/json"
    })
    Call<TMap.SearchPoiResponse> search(@Query("searchKeyword") String query);
}

File: app/src/main/kotlin/me/zipi/navitotesla/api/TMapApi.kt
package me.zipi.navitotesla.api

import me.zipi.navitotesla.model.TMap.SearchPoiResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Query

interface TMapApi {
    @GET("/tmap/pois")
    @Headers("Accept: application/json")
    suspend fun search(@Query("searchKeyword") query: String): Response<SearchPoiResponse>
}